`ultralytics 8.3.151` Add `decimals` argument to Metrics

RizwanMunawar opened the pull request:
## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Improved the `summary()` metric reporting across detection, segmentation, pose, classification, and OBB tasks by adding rounding options, clearer documentation, and more consistent output formatting. 📝✨

### 📊 Key Changes
- Added `normalize` and `decimals` parameters to all `summary()` methods, allowing users to control output precision.
- All metric values in summaries are now rounded to the specified number of decimal places for cleaner, more readable results.
- Expanded and clarified docstrings for each `summary()` method, including usage examples.
- Ensured consistent output structure across detection, segmentation, pose, classification, and OBB metric summaries.
- Updated version to 8.3.151.

### 🎯 Purpose & Impact
- 📈 **Easier to Read Results:** Users get cleaner, more precise metric reports, making it simpler to interpret and share results.
- 🛠️ **Consistent API:** Developers benefit from a unified and predictable summary output across all tasks.
- 📚 **Better Documentation:** Enhanced docstrings and examples help both new and experienced users understand how to use and interpret metric summaries.
- 🎨 **Customization:** Users can now easily adjust the level of detail in their metric outputs to suit their needs.

Overall, this update makes metric reporting in Ultralytics models more user-friendly, customizable, and consistent! 🚀

RizwanMunawar committed:
update DetMetrics

RizwanMunawar committed:
update SegmentMetrics

RizwanMunawar committed:
update PoseMetrics

RizwanMunawar committed:
update ClassifyMetrics

RizwanMunawar committed:
Set normalize=True for metrics

RizwanMunawar committed:
update OBBMetrics

RizwanMunawar committed:
update docsstrings

RizwanMunawar committed:
update docsstrings

RizwanMunawar committed:
Merge 1f123a8f5083feb68d8f2e2b3d5e3f98777f9849 into 9f4eb2491b6a25abaae29b74da6698cefe8dc662

UltralyticsAssistant committed:
Auto-update Ultralytics Docs Reference by https://ultralytics.com/actions

UltralyticsAssistant commented:
👋 Hello @RizwanMunawar, thank you for submitting an `ultralytics/ultralytics` 🚀 PR! To ensure a seamless integration of your work, please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don’t hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

_This is an automated response. An Ultralytics engineer will review your PR and assist you soon!_ 😊

UltralyticsAssistant committed:
Auto-format by https://ultralytics.com/actions

RizwanMunawar committed:
update examples

RizwanMunawar commented:
@glenn-jocher Hi, this PR aligns everything we have for prediction results in metrics. Last week, when we added the support for this feature, `decimals` and `normalize` functionality weren't added (instead, we temporarily added **kwargs.). 

After some testing, I noticed we don't need to use `normalize` as validation metrics are already normalised [0-1], yes, we can use `decimals`, and I added it in the PR. Besides that, I updated the docstrings accordingly. Please let me know what you think. Thanks

RizwanMunawar committed:
update docsstrings

Laughing-q commented on ultralytics/utils/metrics.py:
```diff
@@ -1039,24 +1039,35 @@ def curves_results(self) -> List[List]:
+           decimals (int): Number of decimal places to round the metrics values to.
+
+        Returns:
+           List[Dict]: A list of dictionaries, each representing one class with corresponding metric values.
```
@RizwanMunawar need parentheses here i.e. `(List[Dict])`, same to others

RizwanMunawar committed:
Add (List[Dict]) in docstrings

Laughing-q committed:
Update metrics.py

Laughing-q committed:
Update metrics.py

glenn-jocher committed:
Merge branch 'main' into val-metrics

glenn-jocher committed:
Update __init__.py

Signed-off-by: Glenn Jocher <glenn.jocher@ultralytics.com>

glenn-jocher committed:
Merge branch 'main' into val-metrics
